### ISSUE TITLE
Определить механизм проведения тестирования.

### ISSUE THREAD

**Issue by superhackkiller1997** [i-rinat/c10k-benchmark-fixture#1]
Дать полное описание процесса тестирования. Каким образом и что люди должны мерить, а так же какие результаты должны получить люди и как их интерпретировать.

Пока что есть только куцое описание того, что автор предполагает за тестирование и пожелание "делайте что хотите и измеряйте сами". Т.е. автор посылает людей тестировать, но не описывает того - как они должны это сделать.

**Comment by i-rinat** [comment]
> Дать полное описание процесса тестирования.

Примерное описание дано, этого достаточно.

> Каким образом и что люди должны мерить, а так же какие результаты должны получить люди и как их интерпретировать.

Я не собираюсь рассказывать людям, куда они должны смотреть и что увидеть. Каждый волен принимать решение сам.

**Comment by superhackkiller1997** [comment]
Этого недостаточно. Ты взялся выкатить бенчмарк, назвал это бенчмарком, но это не бенчмарк. Никто не должен догадываться о том - что ты там хотел сказать. Любой должен мочь взять и запустить его. Получив результат. Результата ты не дал, результат почить нельзя.

Тем более - ты там кое что утверждал, что значит бенчмарк - твой пруф. Я понимаю, что на результатах ты прогорел и это было предсказуемо, но всё же. Сделай до конца то, что начал. Ты ведь не бросаешь проекты на полпути? Утверждал - соответствуй.

А по поводу людей - ты не дал им ничего. И вывод никаких сделать нельзя.

**Comment by i-rinat** [comment]
> Ты взялся выкатить бенчмарк

Нет, я просто опубликовал текущий вариант кода. Как видишь, в нём не доделан `server_epoll.c`. И поэтому из него не собирается исполняемый файл.

> назвал это бенчмарком

Нет, даже в названии репозитория написано по-другому. Это benchmark fixture. Приспособление для бенчмарка, а не собственно бенчмарк.

> но это не бенчмарк

Абсолютно верно. Это не бенчмарк.

> Ты ведь не бросаешь проекты на полпути? Утверждал - соответствуй.

Я не собираюсь ни доказывать, ни опровергать приписанные мне утверждения. Это не конструктивно.

> Ах да, по поводу надежд на то, что кто-то придёт и "исправит" епул-версию.

???

---
Возможно, ты считаешь, что этот код — попытка что-то доказать тебе или кому-то ещё. Это не так.

Я изначально описывал проблему в виде реверс-прокси, так как мне это было интересно. Но обсуждения на форуме были о чём угодно, но не об этой задаче. Было несколько вариантов кода, но это всё были эхо-сервера, что по смыслу совсем другое. Задача реверс-прокси никому не интересна? Что ж, путь будет так.

Но она интересна мне, вот я её и ковыряю. Пожалуйста, не выдумывай то, чего нет. Этот код к тебе никак не относится. Хочешь смотреть — смотри. Но вот реализовывать чужие хотелки или нет — решать мне.

**Comment by superhackkiller1997** [comment]
Понимаешь в чём штука. Я уже говорил - если ты действительно вменяем - ты поймёшь то, о чём я тебе говорю и говорил.

Когда ты ставь условия - ты должен осознавать их следствия, если ты их определяешь как опровержение моих доводов - ты должен осознавать причины. Всё просто. И тогда у тебя не будет проблем и не будет того, в чём вы там все оказались.

Вот ты придумал свои условия - зачем? Я знаю зачем ты их придумал и не важно что тебе интересно - изначальная причина не в этом. Если ты интересуешься, если ты не знаешь - зачем ты всё это делаешь? Ты не задавал себе подобный вопрос?

Каждое условие должно быть обосновано. Я тебе говорил, что твои условия на результат не повлияют - твоя задача вообще не имеет какого-либо отношения к производительности.

Не важно что ты там считаешь, что ты там выдумываешь. Никакой "скорости отдачи нет" - есть лок, т.е. слип в обработчике. Если ты этого не понимаешь - пойми, изучи вопрос. Не важно как ты там это называешь.

Тем более ты сам себе противоречишь. Зачем ты пишешь весь этот бред на форуме? Меня в теме нет - можно развекаться? Ну дак это глупо.


И самое главное - почему-то я должен реализовывать ваши хотелки, должен реализовывать твои сливы и постоянные смены условий. А ты ничего не должен - странная логика.


Если ты хочешь получить реализацию/прогрес своей темы - определи чётко условия. Если ты не хочешь - зачем пастишь в паблик, если никто из неё вынести ничего не может?

А по поводу доказательств. Возможно доказать крайне призрачны, когда не знаешь того - что вообще хочешь доказать. Нет смысла кичиться тем, что ты не доказываешь в  условиях, когда сама эта идея тщетна.


Мне так лень идти и что-то писать на форуме, разоблачать и прочее. Мне так жаль, что активные пацаны ничего не понимают, либо делают вид, что ничего не понимают. Надеюсь, что в будущем они смогут вам отвечать.

**Comment by superhackkiller1997** [comment]
В итоге. Хочешь конструктива и какого-то общественно участия в твоих поисках "интересного" - определи внятно условия и реализуй бенчмарк. Я хочу поучаствовать в представлении. Да и думаю, что пацаны так же хотят.

Если тебе это не нужно - там и напиши. Я помогу.
